labgrid/resource/mqtt.py

```diff
@@ -0,0 +1,81 @@
+import queue
+import logging
+from time import monotonic
+
+import attr
+import paho.mqtt.client as mqtt
+import threading
+
+from .common import ManagedResource, ResourceManager, Resource
+from ..factory import target_factory
+from ..util import Timeout
+
+@attr.s(eq=False)
+class MQTTManager(ResourceManager):
+    _available = attr.ib(default=attr.Factory(list), validator=attr.validators.instance_of(list))
+    _avail_lock = attr.ib(default=threading.Lock())
+    _clients = attr.ib(default=attr.Factory(dict), validator=attr.validators.instance_of(dict))
+    _topics = attr.ib(default=attr.Factory(list), validator=attr.validators.instance_of(list))
+    _topic_lock = attr.ib(default=threading.Lock())
```

Comment:
You could probably use `factory=` for these. I'm not sure if maybe type annotations are a better solution for internal attributes.

Reply:
Same as above, but I can switch this to use `factory`.